### ISSUE TITLE
HeavyScript vs. TrueTools

### ISSUE THREAD

**Issue by PackElend** [Heavybullets8/heavy_script#89]
Hi there,
thx again for the great video and h[Truenas Scale - Truecharts Nextcloud - Install & Restore - YouTube @ 20:22](https://www.youtube.com/watch?v=TQlmYxPnSrU).
I would like to finish the conversation we had over there, although it is partly obsolete as it becomes now the official tool as stated in [The Future of TrueCharts | TrueCharts](https://truecharts.org/news/the-future) --> CONGRATULATIONS (hope it not causes extra workload) 
I still feel the conversation hasn't come to proper end, so I do here.

I ask what is the difference to [Backup, Migrations and Restore | TrueCharts](https://truecharts.org/manual/SCALE/guides/backup-restore/) although TrueTool is mentioned, which you say [@ 20:22](https://www.youtube.com/watch?v=TQlmYxPnSrU&t=1222s) is fork/copy of HeavyScript?

Your answer was 
> Truecharts and I will sometimes work together to solve a project, we worked together to solve the backup/restore guide issue, and integrate fixes within the scripts. 
I did most of the coding, so both of the scripts can be used interchangeably with their backup guide.

> The only reason they suggest using Truetool, and not heavyscript is because they "verify" their code. Apparently they trust me to develop code, but they don't trust HeavyScript, but they also use most of HeavyScript features? I don't know dude lol.

> In my opinion, Truetool is a rebrand of HeavyScript.

----- 

Moreover, I asked if you still do a full [snapshot](https://www.truenas.com/docs/core/coretutorials/storage/snapshots) of ix-applications as I can find that in [Backup, Migrations and Restore | TrueCharts](https://truecharts.org/manual/SCALE/guides/backup-restore/)?
I was irritated by the fact their guide still showed the TrueNAS replication options what is not mentioned in your video.

So do you incorporated the data set replication trough TrueNAS itself or do you go 100% with you tool
The only answers in this directions are
> It can do some of the same functions in the Truenas SCALE GUI, but that doesn't mean that it is apart of their project. 

> The only reason you need the truecharts guide, is when you are moving machines, or you are restoring from some sort of remote backup. Theres not really any other reason to do it.

**Comment by heavybullets8** [comment]
Hello,

Thank you for getting in touch and sharing your concerns. I appreciate your engagement with the video and your questions.

Addressing your first question about the difference between Backup, Migrations and Restore | TrueCharts and the TrueTool mentioned at 20:22 in the video, it's important to note that TrueTool used to be a fork/copy of HeavyScript. TrueCharts initially recommended TrueTool over HeavyScript because they had more control over the code and could ensure its quality. However, both scripts were interchangeable with their backup guide due to their similar functionality. TrueCharts has since removed TrueTool, and users are now expected to use HeavyScript.
_This information is included for the benefit of others who might read this._

> I was irritated by the fact their guide still showed the TrueNAS replication options what is not mentioned in your video.

The video focused on my personal method for restoring a malfunctioning Nextcloud instance. I didn't use their backup/restore guide since I believe it's intended for working applications, and I wasn't certain it would work for a broken application. That's why I chose that approach. For backing up and restoring a functioning application, it's best to consult TrueCharts, as they actively maintain the applications and their documentation and have the most up-to-date information.

> So do you incorporate the data set replication through TrueNAS itself or do you go 100% with your tool?

In the video, I demonstrated the exact process I followed without omitting any steps. No backups were needed for that restoration method, just the initial PVC and the malfunctioning application.

I don't create manual snapshots as shown in their guide unless I plan to follow the steps in that guide. I simply use heavyscript -b and nothing else. I don't perform any other redundancy tasks. However, I'm not claiming this is the best approach; for a more up-to-date method, it's advisable to consult TrueCharts.
